libs/vm-modules/tests/unit/math.cpp

```diff
@@ -383,4 +392,111 @@ TEST_F(MathTests, tensor_set_from_string)
   EXPECT_TRUE(gt.AllClose(tensor->GetTensor()));
 }
 
+TEST_F(MathTests, tensor_setup_charge_estimator_min_charge_succeeds)
+{
+  static char const *tensor_setup_src = R"(
+    function main()
+      var tensor_shape = Array<UInt64>(2);
+      tensor_shape[0] = 5u64;
+      tensor_shape[1] = 2u64;
+      var x = Tensor(tensor_shape);
+    endfunction
+  )";
+
+  ASSERT_TRUE(toolkit.Compile(tensor_setup_src));
+  ASSERT_TRUE(toolkit.Run(nullptr, tensor_setup_charge_min)) << stdout.str();
+}
+
+TEST_F(MathTests, tensor_setup_charge_estimator_min_charge_fails)
```

Comment:
It's good that you are using descriptive titles, but try to convey the point of each test. The reason this fails is because the charge limit is exceeded; the test above succeeds, because the charge limit is sufficient. Otherwise whoever reads this in the future needs to hunt for that `- 1` to understand what you are testing, which defeats the point of having titles.